crates/katana/executor/src/blockifier/utils.rs

```diff
@@ -397,3 +488,78 @@ pub(super) fn l2_to_l1_messages_from_exec_info(
+        }
+        starknet_api::deprecated_contract_class::EntryPointType::External => {
+            starknet::core::types::EntryPointType::External
+        }
+        starknet_api::deprecated_contract_class::EntryPointType::L1Handler => {
+            starknet::core::types::EntryPointType::L1Handler
+        }
+    };
+    let call_type = match info.call.call_type {
+        blockifier::execution::entry_point::CallType::Call => starknet::core::types::CallType::Call,
+        blockifier::execution::entry_point::CallType::Delegate => {
+            starknet::core::types::CallType::Delegate
+        }
+    };
+
+    let calls = info.inner_calls.iter().map(function_invocation_from_call_info).collect();
+    let events = info
+        .execution
+        .events
+        .iter()
+        .map(|e| starknet::core::types::OrderedEvent {
+            order: e.order as u64,
+            data: e.event.data.0.iter().map(|d| (*d).into()).collect(),
+            keys: e.event.keys.iter().map(|k| k.0.into()).collect(),
+        })
+        .collect();
+    let messages = info
+        .execution
+        .l2_to_l1_messages
+        .iter()
+        .map(|m| starknet::core::types::OrderedMessage {
+            order: m.order as u64,
+            to_address: (Into::<StarkFelt>::into(m.message.to_address)).into(),
+            from_address: (*info.call.storage_address.0.key()).into(),
+            payload: m.message.payload.0.iter().map(|p| (*p).into()).collect(),
+        })
+        .collect();
+
+    let vm_resources = info.vm_resources.filter_unused_builtins();
+    let get_vm_resource =
+        |name: &str| vm_resources.builtin_instance_counter.get(name).map(|r| *r as u64);
+    let execution_resources = starknet::core::types::ExecutionResources {
+        steps: vm_resources.n_steps as u64,
+        memory_holes: Some(vm_resources.n_memory_holes as u64),
+        range_check_builtin_applications: get_vm_resource(RANGE_CHECK_BUILTIN_NAME),
+        pedersen_builtin_applications: get_vm_resource(HASH_BUILTIN_NAME),
+        poseidon_builtin_applications: get_vm_resource(POSEIDON_BUILTIN_NAME),
+        ec_op_builtin_applications: get_vm_resource(EC_OP_BUILTIN_NAME),
+        ecdsa_builtin_applications: get_vm_resource(SIGNATURE_BUILTIN_NAME),
+        bitwise_builtin_applications: get_vm_resource(BITWISE_BUILTIN_NAME),
+        keccak_builtin_applications: get_vm_resource(KECCAK_BUILTIN_NAME),
+        segment_arena_builtin: get_vm_resource(SEGMENT_ARENA_BUILTIN_NAME),
+    };
+
+    FunctionInvocation {
+        contract_address: (*info.call.storage_address.0.key()).into(),
+        entry_point_selector: info.call.entry_point_selector.0.into(),
+        calldata: info.call.calldata.0.iter().map(|f| (*f).into()).collect(),
+        caller_address: (*info.call.caller_address.0.key()).into(),
+        // See https://github.com/starkware-libs/blockifier/blob/main/crates/blockifier/src/execution/call_info.rs#L167
```

Comment:
what's the link about? is it pointing to the correct code, doesn't seem relevant?

Reply:
Pinned the link to a commit in order to avoid this next time. Basically just justifies why doing an `expect` here is ok